File: src/com/example/Purse.java
package com.example;

public class Purse {
    private int platinum,
                gold,
                electrum,
                silver,
                copper;

    public void displayContents() {
        System.out.println("Purse contents:");
        System.out.println("Platinum: " + platinum);
        System.out.println("Gold: " + gold);
        System.out.println("Electrum: " + electrum);
        System.out.println("Silver: " + silver);
        System.out.println("Copper: " + copper + "\n");
    }

    public void displayTotalGold() {
        int     totalGold = gold,
                totalElectrum = electrum,
                totalSilver = silver,
                totalCopper = copper;

        totalSilver += totalCopper / 10;
        totalCopper %= 10;
        totalElectrum += totalSilver / 5;
        totalSilver %= 5;
        totalGold += totalElectrum / 2;
        totalElectrum %= 2;
        totalGold += platinum * 10;

        System.out.println("The total value in gold is:\n" +
                totalGold + "gp " +
                totalElectrum + "ep " +
                totalSilver + "sp " +
                totalCopper + "cp");
    }

    public Purse(int platinum, int gold, int electrum, int silver, int copper) {
        this.platinum = platinum;
        this.gold = gold;
        this.electrum = electrum;
        this.silver = silver;
        this.copper = copper;
    }

    public Purse() {
        this.platinum = 0;
        this.gold = 0;
        this.electrum = 0;
        this.silver = 0;
        this.copper = 0;
    }

    public int getPlatinum() {
        return platinum;
    }

    public void setPlatinum(int platinum) {
        this.platinum = platinum;
    }

    public int getGold() {
        return gold;
    }

    public void setGold(int gold) {
        this.gold = gold;
    }

    public int getElectrum() {
        return electrum;
    }

    public void setElectrum(int electrum) {
        this.electrum = electrum;
    }

    public int getSilver() {
        return silver;
    }

    public void setSilver(int silver) {
        this.silver = silver;
    }

    public int getCopper() {
        return copper;
    }

    public void setCopper(int copper) {
        this.copper = copper;
    }

    public void addPlatinum(int platinum) {
        this.platinum += platinum;
    }

    public void addGold(int gold) {
        this.gold += gold;
    }

    public void addElectrum(int electrum) {
        this.electrum += electrum;
    }

    public void addSilver(int silver) {
        this.silver += silver;
    }

    public void addCopper(int copper) {
        this.copper += copper;
    }

    public void subtractPlatinum(int platinum) {
        this.platinum -= platinum;
    }

    public void subtractGold(int gold) {
        this.gold -= gold;
    }

    public void subtractElectrum(int electrum) {
        this.electrum -= electrum;
    }

    public void subtractSilver(int silver) {
        this.silver -= silver;
    }

    public void subtractCopper(int copper) {
        this.copper -= copper;
    }

}

File: Python/Include/Purse.py
class Purse:
    def __init__(self):
        self.platinum = 0
        self.gold = 0
        self.electrum = 0
        self.silver = 0
        self.copper = 0

    def get_purse(self):
        print("\nEnter the purse contents")
        self.platinum = int(input("Platinum: "))
        self.gold = int(input("Gold: "))
        self.electrum = int(input("Electrum: "))
        self.silver = int(input("Silver: "))
        self.copper = int(input("Copper: "))

    def display_contents(self):
        print(f"Platinum:\t{self.platinum}")
        print(f"Gold:\t\t{self.gold}")
        print(f"Electrum:\t{self.electrum}")
        print(f"Silver:\t\t{self.silver}")
        print(f"Copper:\t\t{self.copper}")

    def total_gold(self):
        t_copper = self.copper
        t_silver = self.silver + t_copper // 10
        t_copper %= 10
        t_electrum = self.electrum + t_silver // 5
        t_silver %= 5
        t_gold = self.gold + (t_electrum // 2) + (self.platinum * 10)
        t_electrum %= 2

        print(f"Total value of the purse in gold: {t_gold}gp", end=" ")
        if t_electrum != 0:
            print(f"{t_electrum}ep", end=" ")
        if t_silver != 0:
            print(f"{t_silver}sp", end=" ")
        if t_copper != 0:
            print(f"{t_copper}cp")
        else:
            print()     # print a blank line for formatting

    def get_platinum(self):
        return self.platinum

    def set_platinum(self, p):
        self.platinum = p

    def add_platinum(self, p):
        self.platinum += p

    def subtract_platinum(self, p):
        self.platinum -= p

    def get_gold(self):
        return self.gold

    def set_gold(self, g):
        self.gold = g

    def add_gold(self, g):
        self.gold += g

    def subtract_gold(self, g):
        self.gold -= g

    def get_electrum(self):
        return self.electrum

    def set_electrum(self, e):
        self.electrum = e

    def add_electrum(self, e):
        self.electrum += e

    def subtract_electrum(self, e):
        self.electrum -= e

    def get_silver(self):
        return self.silver

    def set_silver(self, s):
        self.silver = s

    def add_silver(self, s):
        self.silver += s

    def subtract_silver(self, s):
        self.silver -= s

    def get_copper(self):
        return self.copper

    def set_copper(self, c):
        self.copper = c

    def add_copper(self, c):
        self.copper += c

    def subtract_copper(self, c):
        self.copper -= c